### PR TITLE
doctests to operators.jl and promotions.jl

### PR DESCRIPTION
**Pull request by KristofferC**

### CODE REVIEW

File: base/operators.jl [hunk 1]
```diff
@@ -250,6 +399,11 @@ scalarmin(x::AbstractArray, y               ) = throw(ArgumentError("ordering is
     identity(x)
 
 The identity function. Returns its argument.
+
+```jldoctest
+julia> identity("Well, what did you expect?")
+"Well, what did you expect?"
```
Comment (ararslan) [inline base/operators.jl]: 😆

File: base/operators.jl [hunk 2]
```diff
@@ -477,6 +683,10 @@ fld1{T<:Integer}(x::T, y::T) = fld(x+y-T(1),y)
     fldmod1(x, y)
 
 Return `(fld1(x,y), mod1(x,y))`.
+
+See also: [`fld1`](@ref), [`mod1`](@ref).
+
+See
```
Comment (ararslan) [inline base/operators.jl]: Missing the rest of the sentence?

Comment (KristofferC) [inline base/operators.jl]: thx fixed

**Review by ararslan — APPROVED** [review]
👍

File: base/operators.jl (hunk 1 above)
Comment (KristofferC) [inline base/operators.jl]: Is this ok or should it be something more boring? :P

File: base/promotion.jl [hunk 3]
```diff
@@ -241,6 +252,20 @@ end
...
+```jldoctest
+julia> 3^5
+243
+
+julia> A = [1 2; 3 4]
+2×2 Array{Int64,2}:
+ 1  2
+ 3  4
+
+julia> A^3
```
Comment (tkelman) [inline base/promotion.jl]: this example's good, could use a corresponding word or two of description that it's the matrix power operator

Comment (KristofferC) [inline base/promotion.jl]: just as a comment after the `A^3`?

Comment (tkelman) [inline base/promotion.jl]: I was thinking in the prose of the docstring

Comment (KristofferC) [inline base/promotion.jl]: Updated

**Comment by KristofferC** [comment]
Any more comments?

File: base/promotion.jl [hunk 4]
```diff
@@ -251,7 +251,7 @@ end
 """
     ^(x, y)
 
-Exponentiation operator.
+Exponentiation operator. If `x is a matrix, computes matrix exponentiatin.
```
Comment (ararslan) [inline base/promotion.jl]: Missing closing backtick, misspelled "exponentiation."

Comment (ararslan) [inline base/promotion.jl]: I'd actually reword the second sentence as, "If `x` is a matrix, computes the matrix exponential."

Comment (KristofferC) [inline base/promotion.jl]: wow I botched that...

Comment (KristofferC) [inline base/promotion.jl]: done

**Comment by ararslan** [comment]
LGTM now.

File: base/promotion.jl [hunk 5]
```diff
@@ -240,7 +251,22 @@ end
 """
     ^(x, y)
 
-Exponentiation operator.
+Exponentiation operator. If `x` is a matrix, computes the matrix exponential.
```
Comment (tkelman) [inline base/promotion.jl]: when I see "matrix exponential" I tend to think `expm`, maybe it's clear enough from context though

Comment (KristofferC) [inline base/promotion.jl]: I agree with you. Will change it.

**Pull request merged**